yarl/_url.py

```diff
@@ -1153,7 +1156,52 @@ def join(self, url):
         # See docs for urllib.parse.urljoin
         if not isinstance(url, URL):
             raise TypeError("url should be URL")
-        return URL(urljoin(str(self), str(url)), encoded=True)
+        other: URL = url
+        scheme = other.scheme or self.scheme
+        parts = {
+            k: getattr(self, k) or ""
+            for k in ("authority", "path", "query_string", "fragment")
+        }
+        parts["scheme"] = scheme
+
+        if scheme != self.scheme or scheme not in uses_relative:
+            return URL(str(other))
+
+        # scheme is in uses_authority as uses_authority is a superset of uses_relative
```

Comment:
Sorry, doesn't this comment mean that the left-side is always True? Shouldn't we just remove it then?

Reply:
other way around - if you got to 1170, 
* scheme is in uses_relative, 
* uses_relative is a subset of uses_authority, 
* therefore the first part of the check in 1171 is always true
 
just kept it kept for comparison with urllib